src/me.rs

```diff
@@ -511,15 +511,14 @@ impl MotionEstimation for DiamondSearch {
     blk_w: usize, blk_h: usize,
     best_mv: &mut MotionVector, lowest_cost: &mut u64
   ) {
-    let bo_adj_h = BlockOffset { x: bo_adj.x >> 1, y: bo_adj.y >> 1 };
     let po = PlaneOffset {
       x: (bo_adj.x as isize) << BLOCK_TO_PLANE_SHIFT >> 1,
       y: (bo_adj.y as isize) << BLOCK_TO_PLANE_SHIFT >> 1,
     };
     for omv in pmvs.iter() {
       if let Some(pmv) = omv {
         let mut predictors = get_subset_predictors::<T>(
-          bo_adj_h,
+          bo_adj,
```

Comment:
I think your fix seems not correct there. Because diamond_me_search( ) requires 1/2 resolution predictors.
Thus, the original (i.e. before your fix) bo_adj_h, i.e. 1/2 resolution, should be passed to get_subset_predictors( ).

Reply:
Please, try "bo_adj_h = BlockOffset { x: bo_adj.x >> 1, y: bo_adj.y >> 1 };" replacing bo_adj. 
Given your first patch, I think that is the right 1/2 bo for use with get_subset_predictors( ).

Reply:
My patch may be wrong, but the purpose was to use the full-res bo, to be consistent with the frame motion vectors:
https://github.com/xiph/rav1e/commit/b8ce66027cf72487fe24cd64880ce4526ff29213

Reply:
Sure, but please double check how other BlockOffset and block sizes are converted, like "mvx_min >> 1, mvx_max >> 1, mvy_min >> 1, mvy_max >> 1, blk_w >> 1, blk_h >> 1,". Those, I think, are 1/2 BlockOffset?

Reply:
Yes, in `diamond_me_search()`, but in the call to `get_subset_predictors()` only use full-res values, then decimate the predictors:

https://github.com/xiph/rav1e/blob/10b9f94e7978095d816331bf54f1ce22362f685f/src/me.rs#L523-L533

Reply:
Oh, right- predictors are down shifted there.
Also, as you figured out, get_subset_predictors( ) should get full-pel inputs, including your fix "bo_adj".

Nice catch on this!